src/OrleansRuntime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -131,27 +131,33 @@ internal class PersistentStreamPullingAgent : SystemTarget, IPersistentStreamPul
             // Setup a reader for a new receiver. 
             // Even if the receiver failed to initialise, treat it as OK and start pumping it. It's receiver responsibility to retry initialization.
             var randomTimerOffset = safeRandom.NextTimeSpan(config.GetQueueMsgsTimerPeriod);
-            timer = base.RegisterTimer(AsyncTimerCallback, QueueId, randomTimerOffset, config.GetQueueMsgsTimerPeriod);
+            timer = RegisterTimer(AsyncTimerCallback, QueueId, randomTimerOffset, config.GetQueueMsgsTimerPeriod);
 
             logger.Info((int) ErrorCode.PersistentStreamPullingAgent_04, "Taking queue {0} under my responsibility.", QueueId.ToStringWithHashCode());
+            return TaskDone.Done;
         }
 
         public async Task Shutdown()
         {
             // Stop pulling from queues that are not in my range anymore.
             logger.Info(ErrorCode.PersistentStreamPullingAgent_05, "Shutdown of {0} responsible for queue: {1}", GetType().Name, QueueId.ToStringWithHashCode());
-            if (timer != null)
+            Task localrecieverInitTask;
+            if ((localrecieverInitTask = Interlocked.Exchange(ref recieverInitTask, null)) != null)
```

Comment:
Don't strictly need it,  it's just a safer pattern, but in retrospect, I see how it could give the impression that this code needs to be thread safe, when it doesn't.  It just needs to be 'Orleans turn' safe.

Will return to previous null check + set null pattern.